apm-agent-plugins/apm-log-shader-plugin/apm-log-shader-plugin-common/src/main/java/co/elastic/apm/agent/log/shader/AbstractLogShadingHelper.java

```diff
@@ -103,21 +111,46 @@ public void stopShading(A originalAppender) {
     }
 
     /**
-     * Checks whether we should skip {@code append()} invocations for log events for the given appender.
+     * Must be called exactly once at the enter to each {@code append()} method (or equivalent) invocation in order to properly
+     * detect nested invocations.
+     * This method checks whether we should skip {@code append()} invocations for log events for the given appender.
      * Log event appends should be skipped if they are replaced by ECS-formatted events, meaning if:
      *  - shading is enabled by configuration AND
      *  - replace is enabled by configuration AND
      *  - there is a shade appender for this appender (there isn't when this appender IS A shade appender or it is not a file appender)
      * @param appender the appender
      * @return true if log events should be ignored for the given appender; false otherwise
      */
-    public boolean shouldSkipAppend(A appender) {
-        return loggingConfiguration.getLogEcsReformatting() == LogEcsReformatting.REPLACE && getOrCreateShadeAppenderFor(appender) != null;
+    public boolean onAppendEnter(A appender) {
+        if (callDepth.isNestedCallAndIncrement()) {
+            // If this is a nested call, never skip, as it means that the skipping decision was already made in the
+            // outermost invocation
+            return false;
+        }
+
+        A shadeAppender = getOrCreateShadeAppenderFor(appender);
```

Comment:
This creates a shade appender even if the reformatting=off. Is that intended?

Reply:
See the issue comment for options of making it dynamic. It won't stay this way for sure. But I thought of an approach of creating always and look at configuration every time an append is called. The main problem with that may be that the file will be created already when creating and starting the appender. It's on my list to look into.